realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -616,11 +617,16 @@ void BaseRealSenseNode::setupStreams()
                                  _encoding);
                 }
 
-                if(_pointcloud && (0 != _pointcloud_publisher.getNumSubscribers()))
+                if(_pointcloud && (0 != _pointcloud_xyzrgb_publisher.getNumSubscribers()))
```

Comment:
I suggest adding another `_pointcloud` variable for the Depth Point Cloud or changing the existing parameter to `_rgbd_pointcloud` and connect it to a new [ROS parameter](https://github.com/intel-ros/realsense/blob/91e10b7e051b46ea6edebc56215f483a3b78a0f5/realsense2_camera/src/base_realsense_node.cpp#L104) in order to enable Depth PC and RGBD PC separately.

In addition, it would be nice to get another [Launch file](https://github.com/intel-ros/realsense/blob/development/realsense2_camera/launch/demo_pointcloud.launch) for the Depth PC.

Reply:
Is there a good reason to even have the _pointcloud flag? Now that both topics publish only when subscribed to, I can't think of a reason to enable/disable them using rosparam (since AFIK that's a static config).

Reply:
Seperating the topics by rosparams prevents advertising and allocating other unwanted topics and ROS resources.